### DISCUSSION TITLE
Nikola with a headless CMS?

### DISCUSSION THREAD

**Discussion by Wysardry** [getnikola/nikola#3697] [Q&A]
I have seen on the Nikola website that it has been used with Dato CMS, Netlify and Contentful, but I haven't been able to find out how to go about using it with another CMS in a local install.

Is there support for REST or GRAPHQL APIs?

I'm especially interested in whether Nikola could be used with [Wagtail CMS](https://wagtail.org/) (which is also built with Python), and if so, how that would be done.

Also, has anyone managed to set up [Front Matter CMS](https://frontmatter.codes/) with Nikola?

**Comment by Kwpolska** [comment]
Nikola is a static site generator. It generates static HTML files based on reST/Markdown/etc files in the filesystem. There is no REST/GraphQL support, and there never will be.

I don’t know if anyone ever used Front Matter CMS. Nikola used to work with [Decap CMS](https://decapcms.org/) (formerly known as Netlify CMS, and still not using the *headless CMS* buzzword), but I’m not sure if this still works

**Reply by Wysardry** [reply]
What I'm actually looking for is a way of using a GUI admin panel with (a local install of) Nikola, along the lines of the one included with Lektor or available as a plugin for Jekyll.

Rather than create one from scratch, I was hoping to be able to repurpose an existing tool. Whatever I used would need to work offline on a Windows 7 laptop, which narrows the options.

**Reply by Kwpolska** [reply]
You’d need to look for some tools yourself, running offline on Windows 7 is probably going to make it difficult.

**Reply by Wysardry** [reply]
I'm currently looking into installing Linux on the Windows 7 laptop or running it from a USB drive.

I do also have a Windows 10 machine that is less portable, plus a Raspberry Pi 4 that doesn't have a dedicated screen.
